recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,178 @@
+        self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+        self._cmake.definitions["WITH_PIC"] = self.options.get_safe("fPIC", False)
+        self._cmake.definitions["WITH_TLS"] = self.options.ssl
+        self._cmake.definitions["WITH_CLIENTS"] = self.options.clients
+        self._cmake.definitions["WITH_BROKER"] = self.options.broker
+        self._cmake.definitions["WITH_APPS"] = self.options.apps
+        self._cmake.definitions["WITH_PLUGINS"] = False
+        self._cmake.definitions["WITH_LIB_CPP"] = self.options.build_cpp
+        self._cmake.definitions["WITH_THREADING"] = self.settings.compiler != "Visual Studio"
+        self._cmake.definitions["WITH_WEBSOCKETS"] = self.options.get_safe("websockets", False)
+        self._cmake.definitions["STATIC_WEBSOCKETS"] = self.options.get_safe("websockets", False) and not self.options["libwebsockets"].shared
+        self._cmake.definitions["DOCUMENTATION"] = False
+        self._cmake.definitions["CMAKE_INSTALL_SYSCONFDIR"] = os.path.join(self.package_folder, "res").replace("\\", "/")
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_passwd", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "MOSQ_LIBS", "CONAN_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "include", "mosquitto.h"), "__declspec(dllimport)", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "cpp", "mosquittopp.h"), "__declspec(dllimport)", "")
+        # dynlibs for apple mobile want code signatures and that will not work here
+        # this would actually be the right patch for static builds also, but this would have other side effects, so
+        if(self.settings.os in ["iOS", "watchOS", "tvOS"]):
```

Comment:
```suggestion
        if(self.settings.os in ["iOS", "watchOS", "tvOS"]):
```

Is it true for MacOS? Just curious.

Reply:
no, macos not, just for mobile you need to have a signature step inbuilt in your build